src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -908,7 +908,7 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
   const gboolean download_only =
     ((self->flags & RPMOSTREE_TRANSACTION_DEPLOY_FLAG_DOWNLOAD_ONLY) > 0);
   /* Mainly for the `install` and `override` commands */
-  const gboolean no_pull_base =
+  const gboolean no_pull_base = cache_only ||
```

Comment:
> I think instead we want to have --cliwrap actually turn on no-pull-base if no other options are provided?

Yep, done.

> (Though it'd probably be cleaner UX and implementation-wise if this were a separate ex command instead.)

A whole new DBus API too, like `Initramfs`?  A lot more scaffolding there, and also means we can't easily change it as part of one transaction.  (A flaw in this is that we should really have `rpm-ostree reset -a` also undo `ex-cliwrap` for example too)

Reply:
Oh no, it'd still use `UpdateDeployment`, but at least we wouldn't have that special `argv[]` handling for example. But cool as is too. I understand wanting to avoid boilerplate. :)

Reply:
Yeah, agree on argv handling but what led me here too is this PR also fixes it so `--register-deriver` no longer needs a superfulous empty string and will also no longer try to talk to the network.